Std/Data/HashMap/Basic.lean

```diff
@@ -29,18 +29,18 @@ def mk (buckets := 8) (h : 0 < buckets := by decide) : Buckets α β :=
   ⟨mkArray buckets .nil, by simp [h]⟩
 
 /-- Update one bucket in the bucket array with a new value. -/
-def update (data : Buckets α β) (i : USize)
+def update (data : Buckets α β) (i : UInt64)
     (d : AssocList α β) (h : i.toNat < data.1.size) : Buckets α β :=
-  ⟨data.1.uset i d h, (Array.size_uset ..).symm ▸ data.2⟩
+  ⟨data.1.set ⟨i.toNat, h⟩ d, (Array.size_set ..).symm ▸ data.2⟩
 
 /--
 The number of elements in the bucket array.
 Note: this is marked `noncomputable` because it is only intended for specification.
 -/
-noncomputable def size (data : Buckets α β) : Nat := .sum (data.1.data.map (·.toList.length))
+def size (data : Buckets α β) : Nat := .sum (data.1.data.map (·.toList.length))
```

Comment:
per comment in code this was noncomputable to discourage use outside of specification. does it have to be computable?

Reply:
Oops, sorry, this was from when I was working out what was going on. I'll revert.